### PR TITLE
Refactor code into new API function rz_flag_unset_all_in_space

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr refactors some code into a new API function `rz_flag_unset_all_in_space` to simplify the original code.

**Test plan**

All builds are green.

**Closing issues**

...

### CODE REVIEW

**Review by Maijin — APPROVED** [review]

File: librz/flag/flag.c [hunk 1]
```diff
@@ -841,6 +841,19 @@ RZ_API void rz_flag_unset_all(RzFlag *f) {
 	new_spaces(f);
 }
 
+/* unset all flag items in the space with the given name */
```
Comment (XVilka) [inline librz/flag/flag.c]: Please add a doxygen documentation for all new RZ_API functions.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/flag/flag.c (hunk 1 above)
Comment (kazarmy) [reply librz/flag/flag.c]: Okay f77155b8be4fe0533b15ed5a4e51dcc9eefd5e72

File: librz/flag/flag.c [hunk 2]
```diff
@@ -841,7 +841,12 @@ RZ_API void rz_flag_unset_all(RzFlag *f) {
 	new_spaces(f);
 }
 
-/* unset all flag items in the space with the given name */
+/**
+ * @brief Unset all flag items in the space with the given name
```
Comment (ret2libc) [inline librz/flag/flag.c]: use `\brief` `\param`, etc., like what's done in most other places. See cmd_api.c for example.

Comment (kazarmy) [reply librz/flag/flag.c]: ok cfbd43da16da887f53e113d1f19b6da51d6ecefd

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**